thales_remote/file_interface.py

```diff
@@ -26,277 +26,328 @@
         By default, the files remain in the object.
-        
+
         :param enable: True = Allow the files to remain in the object.
         """
-        self._keepFilesInObject = enable
+        self._keep_files_in_object = enable
         return
-    
-    def disableKeepReceivedFilesInObject(self):
-        """ Disable that the files remain in the Python object.
+
+    def disableKeepReceivedFilesInObject(self) -> None:
+        """
+        disable that the files remain in the Python object
         """
         return self.enableKeepReceivedFilesInObject(False)
-    
-    def getReceivedFiles(self):
-        """ Read all files from the Python object.
-        
+
+    def getReceivedFiles(self) -> list[dict[str, Union[str, bytearray]]]:
+        """
+        read all files from the Python object
+
         This function returns an array. Each element in the array is a dictionary as described in
         function :func:`~thales_remote.file_interface.ThalesFileInterface.aquireFile`.
-        
+
         :returns: Array with the files from the Python object.
-        :rtype: array 
         """
         return self.receivedFiles
-    
-    def getLatestReceivedFile(self):
-        """ Read the latest files from the Python object.    
-        
-        :returns: The latest file from the Python object.
-        :rtype: array 
-        """
-        return  self.receivedFiles[-1]
-    
-    def deleteReceivedFiles(self):
-        """ Delete all files from the Python object.
+
+    def getLatestReceivedFile(self) -> dict[str, Union[str, bytearray]]:
+        """
+        read the latest files from the Python object
+
+        :returns: latest file from the Python object
+        """
+        return self.receivedFiles[-1]
+
+    def deleteReceivedFiles(self) -> None:
+        """
+        delete all files from the Python object
         """
         self.receivedFiles = []
         return
-        
-    """
-    The following methods should not be called by the user.
-    They are marked with the prefix '_' after the Python convention for proteced.
-    """
-    
-    def _saveReceivedFile(self, fileToWrite):
-        """ Saves the passed file to disk.
-        """
-        if self._saveReceivedFilesToDisk == True:
-            fileNameWithPath = os.path.join(self.pathToSave,fileToWrite["name"])
-            
-            file = open(fileNameWithPath,"wb")
-            file.write(fileToWrite["binary_data"])
-            file.close()
+
+    # The following methods should not be called by the user.
+    # They are marked with the prefix '_' after the Python convention for proteced.
+
+    def _saveReceivedFile(self, fileToWrite: str) -> None:
+        """
+        saves the passed file to disk ONLY IF self._save_received_files_to_disk is enabled
+
+        TODO:
+            - if not self._save_received_files_to_disk, then this method seems to fail silently
```

Comment:
this function is also only called by the receive function if self._save_received_files_to_disk is True, so this is double checked and i can remove the internal one.